packages/tools/src/utilities/pointInSurroundingSphereCallback.ts

```diff
@@ -4,33 +4,134 @@ import type { Types } from '@cornerstonejs/core';
-// is the center of the sphere ..
-
 /**
- * Given a viewport, an imageData, and a circle points in the viewport, it will
- * run the callback for each point in sphere whose great circle (biggest circle
- * that can be drawn chopping a sphere) is the provided circle points.
+ * Given an imageData, and the great circle top and bottom points of a sphere (
+ * great circle also known as orthodrome is the intersection of the sphere and
+ * the plane that passes through the center of the sphere), this function will
+ * run the callback for each point of the imageData that is within the sphere
+ * defined by the great circle points.
  *
- * @param viewport - VolumeViewport
  * @param imageData - The volume imageData
- * @param circlePoints - [Types.Point3, Types.Point3]
+ * @param circlePoints - bottom and top points of the great circle in world coordinates
  * @param callback - A callback function that will be called for each point in the shape.
  */
 export default function pointInSurroundingSphereCallback(
-  viewport: Types.IVolumeViewport,
   imageData: vtkImageData,
   circlePoints: [Types.Point3, Types.Point3],
-  callback: PointInShapeCallback
+  callback: PointInShapeCallback,
+  viewport?: Types.IVolumeViewport
 ): void {
+  // We can run the sphere equation to determine if a point is inside
+  // the sphere; however, since the imageData dimensions can be quite large, we
+  // can narrow down the search by estimating the bounds of the sphere in index
+  // space.
+  const { boundsIJK, centerWorld, radiusWorld } = _getBounds(
+    circlePoints,
+    imageData,
+    viewport
+  );
+
+  const sphereObj = {
+    center: centerWorld,
+    radius: radiusWorld,
+  };
+
+  pointInShapeCallback(
+    imageData,
+    (pointLPS) => pointInSphere(sphereObj, pointLPS),
+    callback,
+    boundsIJK
+  );
+}
+
+function _getBounds(
+  circlePoints: [Types.Point3, Types.Point3],
+  imageData: vtkImageData,
+  viewport
+): {
+  boundsIJK: BoundsIJK;
+  centerWorld: Types.Point3;
+  radiusWorld: number;
+} {
+  const [bottom, top] = circlePoints;
+
+  // Sphere center in world
+  const centerWorld = vec3.fromValues(
+    (bottom[0] + top[0]) / 2,
+    (bottom[1] + top[1]) / 2,
+    (bottom[2] + top[2]) / 2
+  );
+
+  // sphere radius in world
+  const radiusWorld = vec3.distance(bottom, top) / 2;
+
+  let boundsIJK;
+
+  if (!viewport) {
+    // If no viewport is provide (no camera), we can estimate the bounding box
+    // of the sphere in index space.
+    // This is done by calculating the maximum value for radius in the index
+    // space (since the radius is in world space, we need to convert it to index, and
+    // each dimensions can have a different scale factor). Therefore, by finding
+    // the minimum spacing value in the imageData, we can calculate the maximum
+    // radius in index space and use that to calculate the bounds of the sphere
+    // This will not be accurate, but it is a good first approximation.
```

Comment:
Considering one of the dimensions can be 2-10x bigger than the others, I'm not sure this is a good approximation?